tests/ignite/distributed/utils/__init__.py

```diff
@@ -119,6 +119,27 @@ def _test_distrib_all_reduce(device):
         assert res.device == t.device, f"{res.device} vs {t.device}"
 
 
+def _test_distrib_all_reduce_group(device):
+
+    if idist.get_world_size() > 1 and idist.backend() is not None:
+        ranks = [0, 1]
+        rank = idist.get_rank()
+        t = torch.tensor([rank], device=device)
+
+        group = idist.new_group(ranks)
+        res = idist.all_reduce(t, group=group)
+        assert res == torch.tensor([sum(ranks)])
+
+        t = torch.tensor([rank], device=device)
+        res = idist.all_reduce(t, group=ranks)
+        assert res == torch.tensor([sum(ranks)])
+
+        ranks = "abc"
+        with pytest.raises(ValueError, match=r"Argument ranks should be list of int"):
+            group = idist.new_group(ranks)
+            res = idist.all_reduce(t, group=group)
```

Comment:
test cases should be: 
- group arg as ranks [0, 1]
- group arg as a backend group created with `idist.new_group`
- invalid group arg like "abc"